packages/rx-utils/src/rd/api-client.utils.ts

```diff
@@ -0,0 +1,91 @@
+import { RemoteData } from '@devexperts/remote-data-ts';
+import * as qs from 'query-string';
+import { of } from 'rxjs/observable/of';
+import { ajax } from 'rxjs/observable/dom/ajax';
+import { pending, failure, success } from '@devexperts/utils/dist/rd';
+import { AjaxRequest } from 'rxjs/observable/dom/AjaxObservable';
+import { Observable } from 'rxjs/Observable';
+import { Subject } from 'rxjs/Subject';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/catch';
+import 'rxjs/add/operator/startWith';
+
+export enum RequestMethod {
+	Get = 'GET',
+	Post = 'POST',
+	Put = 'PUT',
+	Delete = 'DELETE',
+}
+
+export class ApiClient<TError> {
+	static create<T>(baseHref: string, extraHeaders?: object) {
+		return new ApiClient<T>(baseHref, extraHeaders);
+	}
+
+	readonly RequestMethod = RequestMethod;
+
+	private constructor(private readonly baseHref: string, readonly headers?: object) {}
+
+	private createURL(url: string, query?: {}): string {
+		return query ? `${url}?${qs.stringify(query)}` : url;
+	}
+
+	private readonly errorSubj$ = new Subject<TError>();
+
+	readonly error$ = this.errorSubj$.asObservable();
+
+	readonly request = <Response = never>(request: AjaxRequest): Observable<RemoteData<TError, Response>> => {
```

Comment:
`ApiClient#request` actually emits only `AjaxError` errors, there's no need for error generalization.

Reply:
done